File: source/Mlos.Python/mlos/MlosOptimizationServices/ModelsDatabase/Relations/Optimizer.py
#
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT License.
#
class Optimizer:
    """ An instance of optimizer in the ModelsDatabase

    """
    def __init__(
            self,
            optimizer_id=None,
            optimizer_type=None,
            optimizer_config=None,
            optimizer_hypergrid=None,
            optimizer_focused_hypergrid=None,
            optimization_problem=None,
            models=None,
            registered_params_combos=None
    ):
        self.optimizer_id = optimizer_id
        self.optimizer_type = optimizer_type
        self.optimizer_config = optimizer_config
        self.optimizer_hypergrid = optimizer_hypergrid
        self.optimizer_focused_hypergrid = optimizer_focused_hypergrid
        self.optimization_problem = optimization_problem
        self.models = models if models is not None else []
        self.registered_params_combos = registered_params_combos

File: source/Mlos.Model.Services/ModelsDb/ObjectRelationalMappings/Optimizer.cs
// -----------------------------------------------------------------------
// <copyright file="Optimizer.cs" company="Microsoft Corporation">
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See LICENSE in the project root
// for license information.
// </copyright>
// -----------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Text;

namespace Mlos.Model.Services.ModelsDb.ObjectRelationalMappings
{
    /// <summary>
    /// Maps the optimizer object to its representation in the ModelsDatabase.
    /// </summary>
    public class Optimizer
    {
        public enum RemoteOptimizerType
        {
            SimpleBayesianOptimizer,
        }

        /// <summary>
        /// Indicates the type of the remote optimizer. As we add more optimizers and evolve them
        /// this will have to become more complex: include versioning information, etc..
        /// </summary>
        public RemoteOptimizerType OptimizerType { get; set; }

        /// <summary>
        /// Optimization problem represented as a JsonString.
        /// </summary>
        public string OptimizationProblemJsonString { get; set; }

        public Guid? OptimizerId { get; set; }
    }
}
